gdal/gcore/overview.cpp

```diff
@@ -1453,21 +1433,25 @@ GDALResampleChunk32R_AverageT( double dfXRatioDstToSrc,
                     for( int iX = nSrcXOff; iX < nSrcXOff2; ++iX )
                     {
                         const T val = pChunk[iX + static_cast<GPtrDiff_t>(iY) *nChunkXSize];
-                        const unsigned nVal = static_cast<unsigned>(val);
-                        if( nVal < static_cast<unsigned>(nEntryCount) && aEntries[nVal].c4 )
+                        // cppcheck-suppress unsignedLessThanZero
```

Comment:
There's nothing there to guarantee that the val, of type T, would be positive:
```
  double d = -1234;
  unsigned val = (unsigned)d;
```
isn't going to give you the result you want.

Reply:
> There's nothing there to guarantee that the val, of type T, would be positive:

hum, yes although that it would mean that we have a GDAL band of type Double and with a ColorTable. This is not possible with TIFF for example, but yes, could probably be possible with other formats (at least by crafting some VRT). And that wouldn't really make sense, as a negative value couldn't match an entry in the table.
OK, let's keep your change.

Reply:
You have to know more to figure that out.

From the *local* code, all you know is that some value is going to be used as an index to a vector so needs to be guaranteed in range. The compiler will eliminate the check if it's not necessary for the instantiation anyway.